src/Toolkit/Toolkit.UI.Controls/MeasureToolbar/MeasureToolbar.cs

```diff
@@ -559,144 +590,39 @@ public MapView? MapView
-            toolbar.DisplayResult(newMapView.SketchEditor?.Geometry);
-        }
-
-        /// <summary>
-        /// Gets or sets the sketch editor used for measuring distances.
-        /// </summary>
-        public SketchEditor? LineSketchEditor
-        {
-            get { return GetValue(LineSketchEditorProperty) as SketchEditor; }
-            set { SetValue(LineSketchEditorProperty, value); }
-        }
-
-        /// <summary>
-        /// Identifies the <see cref="LineSketchEditor"/> dependency property.
-        /// </summary>
-        public static readonly DependencyProperty LineSketchEditorProperty =
-            DependencyProperty.Register(nameof(LineSketchEditor), typeof(SketchEditor), typeof(MeasureToolbar), new PropertyMetadata(null, OnLineSketchEditorPropertyChanged));
-
-        private static void OnLineSketchEditorPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
-        {
-            var toolbar = (MeasureToolbar)d;
-            var newSketchEditor = e.NewValue as SketchEditor;
-            if (newSketchEditor == null)
-            {
-                throw new ArgumentException($"{nameof(LineSketchEditor)} cannot be null or empty.");
-            }
-
-            var oldSketchEditor = e.OldValue as SketchEditor;
-            if (oldSketchEditor != null)
-            {
-                oldSketchEditor.GeometryChanged -= toolbar.OnGeometryChanged;
-            }
-
-            newSketchEditor.GeometryChanged += toolbar.OnGeometryChanged;
-            toolbar.DisplayResult(newSketchEditor.Geometry);
-        }
-
-        /// <summary>
-        /// Gets or sets the sketch editor used for measuring areas.
-        /// </summary>
-        public SketchEditor? AreaSketchEditor
-        {
-            get { return GetValue(AreaSketchEditorProperty) as SketchEditor; }
-            set { SetValue(AreaSketchEditorProperty, value); }
-        }
-
-        /// <summary>
-        /// Identifies the <see cref="AreaSketchEditor"/> dependency property.
-        /// </summary>
-        public static readonly DependencyProperty AreaSketchEditorProperty =
-            DependencyProperty.Register(nameof(AreaSketchEditor), typeof(SketchEditor), typeof(MeasureToolbar), new PropertyMetadata(null, OnAreaSketchEditorPropertyChanged));
-
-        private static void OnAreaSketchEditorPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
-        {
-            var toolbar = (MeasureToolbar)d;
-            var newSketchEditor = e.NewValue as SketchEditor;
-            if (newSketchEditor == null)
-            {
-                throw new ArgumentException($"{nameof(AreaSketchEditor)} cannot be null or empty.");
-            }
-
-            var oldSketchEditor = e.OldValue as SketchEditor;
-            if (oldSketchEditor != null)
+            toolbar._geometryEditor = newMapView.GeometryEditor;
+            if (toolbar._geometryEditor != null)
```

Comment:
Do these lines on measure start instead, and unhook on measure completion / cancellation (should also clear up hooks if I call `mapView.GeometryEditor.Cancel()` externally).